### DISCUSSION TITLE
Database returns undefined even though data exists

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [supabase/supabase#5703] [Questions] [answered]
I am trying to query the database likes this
```
const {chapters, error} = await supabase
    .from('chapters')
    .select('id,chapternum,title,createdat,is_published')
    .eq('seriesid', seriesid)
    .order('chapternum', { ascending: true })
```

The column `seriesid` has a foreign key relationship but I am not accessing the data through the foreign key, I am only using it to query the results.

When I log the results, it shows `undefined`

**Comment by GaryAustin1** [comment, marked as answer]
I think you want const {data:chapters, error} =....  if you mean chapters is undefined.

**Reply by Electric-Dragon** [reply]
Yes it worked, thanks a lot! But I do not get why I have to do `{data:chapters, error}` instead of only `{data, error}`. Can you help me understand?

**Reply by GaryAustin1** [reply]
You can do {data, error} but then your result is in variable data.  You showed {chapters, error} which won't work.  By default the names are data and error.  If you want something different you have to do data:newname.

**Reply by Electric-Dragon** [reply]
Ohh okay...thanks a lot!
